Common/include/CConfig.hpp

```diff
@@ -7384,6 +7596,27 @@ class CConfig {
    */
   void SetSurface_TotalPressure(unsigned short val_marker, su2double val_surface_totalpressure) { Surface_TotalPressure[val_marker] = val_surface_totalpressure; }
 
+  /*!
+   * \brief Set the CO mass fraction at the surface.
+   * \param[in] val_imarker - Index corresponding to the outlet boundary.
+   * \param[in] val_surface_co - Value of the CO mass fraction.
+   */
+  void SetSurface_CO(unsigned short val_imarker, su2double val_surface_co){ Surface_CO[val_imarker] = val_surface_co; };
+
+  /*!
+   * \brief Set the NOx mass fraction at the surface.
+   * \param[in] val_imarker - Index corresponding to the outlet boundary.
+   * \param[in] val_surface_no - Value of the NOx mass fraction.
+   */
+  //void SetSurface_Scalar(unsigned short val_imarker, su2double val_surface_scalar, unsigned short val_i_scalar){ Surface_Scalar[val_imarker][val_i_scalar] = val_surface_scalar; };
+
+  /*!
+   * \brief Set the NO mass fraction at the surface.
+   * \param[in] val_imarker - Index corresponding to the outlet boundary.
+   * \param[in] val_surface_no - Value of the NO mass fraction.
+   */
+  void SetSurface_NOx(unsigned short val_imarker, su2double val_surface_nox){ Surface_NOx[val_imarker] = val_surface_nox; };
```

Comment:
Do you use this "Set" or was it implemented for completeness?
We should try to keep the config as read-only as possible, and we should especially avoid using it to store things that are solution variables (like we do with AoA in fixed CL mode, it is as bad as using global variables).

Reply:
yes, in combustion we solve usually 4 transport equations: progress variable (PV), enthalpy,carbon monoxide CO and nitrogen oxides NOx. Our main objective is design optimization to minimize CO and NOx.